python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -262,6 +263,18 @@ def get_partition_keys_in_range(self, partition_key_range: PartitionKeyRange) ->
         start_time = self.start_time_for_partition_key(partition_key_range.start)
         end_time = self.start_time_for_partition_key(partition_key_range.end)
 
+        partition_keys = self.get_partition_keys()
```

Comment:
Calling `get_partition_keys` is expensive and makes an operation that's otherwise O(size of range), O(size of partition set). Is it possible to avoid it?